### DISCUSSION TITLE
ValueError "User.config" must be a "UserConfig" instance

### DISCUSSION THREAD

**Discussion by MakoWish** [netbox-community/netbox#17742] [Help Wanted!]
I would appreciate some help with this. A colleague at work pointed me to NetBox, and I wanted to test it out. I am attempting to get up and running in my lab, and at the point in the documentation where we test via `python3 manage.py runserver 0.0.0.0:8000 --insecure`, I get a login page, but after entering the credentials I set, I get the error below. 

# Server Error

There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'ValueError'>

Cannot assign "<netbox.config.Config object at 0x7ace1d5d3680>": "User.config" must be a "UserConfig" instance.

Python version: 3.12.3
NetBox version: 4.1.3
Plugins: None installed

If further assistance is required, please post to the [NetBox discussion forum](https://github.com/netbox-community/netbox/discussions) on GitHub.
[Home Page](http://foo.bar.lab:8000/)

**Comment by KriKrixs** [comment]
Hi @MakoWish 

I had the same issue and resolved it by deleting the database in Postgre and recreate it using the documentation.

**Reply by termark** [reply]
Thank you, this worked for me as well

**Comment by MakoWish** [comment]
Thank you, @KriKrixs . Since this was a new build in a lab environment, I just blew the entire VM away and started from scratch last night. I am no longer getting that error. Now... to get LDAP authentication to work! Hahaha

**Reply by KriKrixs** [reply]
Good luck mate 😂, have a nice day !

**Reply by MakoWish** [reply]
I actually got it working just a few minutes later. I had mistyped a DN. I should have copy/pasted it in the first place.

**Comment by lgt-bu** [comment]
Hi, I recently upgraded from 3.6.x to 4.1.6 (i did requisite path from 3.6.x to 3.7.8, then onwards to 4.1.6) and since then, adding a new user we get that message above when they try to login.

On our system:

```
Server Error

There was a problem with your request. Please contact an administrator.

The complete exception is provided below:

<class 'ValueError'>

Cannot assign "<netbox.config.Config object at 0x7f41c868cdf0>": "User.config" must be a "UserConfig" instance.

Python version: 3.10.0
NetBox version: 4.1.6
Plugins: 
  netbox_reorder_rack: 1.1.3
  netbox_topology_views: 4.1.0

If further assistance is required, please post to the [NetBox discussion forum](https://github.com/netbox-community/netbox/discussions) on GitHub.
[Home Page](https://netbox.ourserver.uk)
```

I'm not confident with a resolution of just starting again. Nor with the idea that I just delete the database. Where is that solution and how safe? We are not using the ldap plugin. It's just local accounts on the box.

**Reply by dbrummer** [reply]
I noticed a similar issue with trying to add an additional superuser via the Helm chart values.yaml.  It created the user but didn't create the required entry in `public.users_userconfig`.  I had to manually enter a new row with the `data` column set to `{}` and `user_id` as the `id` of the newly created user.  This is was on `NetBox Community v4.2.4-Docker-3.2.0` and I'm looking into a potential fix.

**Reply by maxabaumgarten** [reply]
ran into the same thing. your fix works


NetBox release | NetBox Community v4.2.5-Docker-3.2.0 (2025-03-06)
Python Version: 3.12.3
Django Verison 5.1.7
PostgreSQL 17.3 on aarch64-unknown-linux-gnu

**Comment by Guruleenyc** [comment]
I just experienced this error for a new docker installation on Unraid. The error occurs when I try to login to Netbox. How can I resolve?

**Comment by Fubs** [comment]
@dbrummer's fix worked for me. If you can get into the postgres database then you should be able to fix with the query `INSERT INTO users_userconfig VALUES (0, '{}', 1);`.

I'm running this docker compose file

```
$ cat docker-compose.yml 
---
services:
  netbox:
    image: lscr.io/linuxserver/netbox:latest
    container_name: netbox
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=<my timezone>
      - SUPERUSER_EMAIL=<my email>
      - SUPERUSER_PASSWORD=<my superuser password>
      - ALLOWED_HOST=<my allowed host>
      - DB_NAME=netbox
      - DB_USER=netbox
      - DB_PASSWORD=<password for netbox postgres db>
      - DB_HOST=postgres.myserverhostname.local
      - DB_PORT=5432
      - REDIS_HOST=redis.myserverhostname.local
      - REDIS_HOST=6379
      - REDIS_DB_TASK=5
      - REDIS_DB_CACHE=6
    volumes:
      - /storage_ssd/netbox/config:/config
    ports:
      - 33006:8000
    restart: unless-stopped
```


Before starting netbox for the first time, I made a postgres role called "netbox" and a database called "netbox" owned by the new netbox role. Then I ran `docker compose up`, let it populate the postgres database and make the config files, then stopped the docker compose service. Then, I connected to the database with
```
$ psql -U postgres -h postgres.myserverhostname.local netbox
```

ran the insert query

```
psql (17.5)
Type "help" for help.

netbox=# INSERT INTO users_userconfig VALUES (0, '{}', 1);
```

restarted the docker compose service, and after it started up I was able to login with the username `admin` and the `<my superuser password>` from the docker compose file.

**Comment by elixdreamer** [comment]
Hello,

Did encounter the same in NetBox Community v4.4.4-Docker-3.4.1

running the following script to resolve this in nbshell, see wiki on how to get in the Netbox Shell (nbshell).

```
from users.models import User, UserConfig

# Fix all users missing config
for user in User.objects.all():
    try:
        # Try to access config
        _ = user.config
        print(f"✓ User '{user.username}' has valid config")
    except User.config.RelatedObjectDoesNotExist:
        # Create missing UserConfig
        UserConfig.objects.create(user=user)
        print(f"✓ Created config for user '{user.username}'")

print("\nAll users fixed!")
```

**Reply by ijumadurdyyew** [reply]
It's working!!! Thank you!

For all other people seeking information about how to fix it. DON'T delete DB, this is very very bad habbit
